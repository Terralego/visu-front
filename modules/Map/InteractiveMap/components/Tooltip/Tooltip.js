import React from 'react';

import MarkdownRenderer from '../../../../Template/MarkdownRenderer';
import FeatureProperties from '../../../FeatureProperties';

export const Tooltip = ({ fetch, properties, ...props }) => (
  <FeatureProperties
    {...fetch}
    properties={properties}
  >
    {allProperties => (
      <MarkdownRenderer
        {...props}
        {...allProperties}
      />
    )}
  </FeatureProperties>
);

export default Tooltip;
