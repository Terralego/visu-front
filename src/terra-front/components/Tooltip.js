import React from 'react';

import { Tooltip as BpTooltip } from '@blueprintjs/core';
import withDeviceSize from '../hoc/withDeviceSize';

export const Tooltip = ({ isMobileSized, children, ...props }) => (
  isMobileSized
    ? children
    : (
      <BpTooltip {...props}>
        {children}
      </BpTooltip>
    )
);
export default withDeviceSize()(Tooltip);
