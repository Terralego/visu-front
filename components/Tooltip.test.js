import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from '@blueprintjs/core';
import Tooltip from './Tooltip';

jest.mock('@blueprintjs/core', () => ({
  Button: () => <p>Button</p>,
}));

it('should render with desktop device size', () => {
  const props = { content: 'foo' };
  const tree = renderer.create((
    <Tooltip
      isMobileSized={false}
      {...props}
    >
      <Button />
    </Tooltip>
  ));
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render with tablet device size', () => {
  const tree = renderer.create((
    <Tooltip
      isMobileSized
    >
      <Button />
    </Tooltip>
  ));
  expect(tree.toJSON()).toMatchSnapshot();
});
