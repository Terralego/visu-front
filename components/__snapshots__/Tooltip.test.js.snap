// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with desktop device size 1`] = `
<p>
  Tooltip
</p>
`;

exports[`should render with tablet device size 1`] = `
<p>
  Tooltip
</p>
`;
